main.go

```diff
@@ -129,23 +149,60 @@ func main() {
 		}
 	}
 
+	// Set retryable HTTP client.
+	clientFetcher.Transport = NewRetryableTransport(&RetryableTransportCfg{
+		Transport:       clientFetcher.Transport,
+		InitialInterval: 200 * time.Millisecond,
+		MaxInterval:     2 * time.Second,
+		MaxElapsedTime:  10 * time.Second,
+	})
+
 	var f fetcher
+	var gr run.Group
+	var tenset tenantsSetter
 
+	// If rulesBackendURL is specified, use it to fetch rules in priority.
+	// Otherwise, use observatoriumURL to fetch rules.
 	if cfg.rulesBackendURL != "" {
-		rulesFetcher, err := newRulesBackendFetcher(cfg.rulesBackendURL, clientFetcher)
-		if err != nil {
-			log.Fatalf("failed to initialize Rules Backend fetcher: %v", err)
+		rof := configureRulesObjtoreFetcher(cfg, clientFetcher)
+		tenset = rof
+
+		// If at least one tenant is specified, use GetTenantsRules to fetch rules for each tenant.
+		// Otherwise, use GetAllRules to fetch rules for all tenants.
+		if len(cfg.tenant) > 0 {
+			f = fetcherFunc(rof.GetTenantsRules)
+		} else {
+			f = fetcherFunc(rof.GetAllRules)
 		}
```

Comment:
Could simplify this logic by initializing `f` with `rof.GetAllRules`, for example, and replacing if with the tenant aware one if tenants are configured.

Reply:
Yup, I agree, changing that.